eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/AbstractEventsourcedView.scala

```diff
@@ -151,6 +153,20 @@ abstract class AbstractEventsourcedView(val id: String, val eventLog: ActorRef)
       "Use commandContext.become(...) to change it.")
 
   /**
+   * Java API that sets this actor's [[EventsourcedView.onCommand command]] handler.
+   *
+   * Supplied with a partial function that defines the actor's command handling behaviour.
+   * Use [[$akkaReceiveBuilder ReceiveBuilder]] to define the behaviour.
+   *
+   * If [[onCommand]] is implemented, the supplied behaviour is ignored.
+   *
+   * @param handler a function that defines this actor's command handling behaviour.
+   * @see [[EventsourcedView]]
+   */
+  final protected def setOnCommand(handler: AbstractActor.Receive): Unit =
```

Comment:
To keep the existing Eventuate-API pattern but also align with the new Akka-Java API I would suggest to offer two variants of each handler - a `setOnXX` and a `createOnXX` (which mimics the new Akka-Java API).

This could look something like this
```
private var commandBehaviour: AbstractActor.Receive = AbstractActor.emptyBehavior

def createOnCommand(): AbstractActor.Receive =
  AbstractActor.emptyBehavior
  
final protected def setOnCommand(handler: AbstractActor.Receive): Unit =
  if (commandBehaviour == AbstractActor.emptyBehavior) commandBehaviour = handler
  else throw new IllegalStateException("Actor command behaviour has already been set with setOnCommand(...).  " +
      "Use commandContext.become(...) to change it.")

final override def onCommand =
  createOnCommand().orElse(commandBehaviour).onMessage.asInstanceOf[Receive]
```

This way the `setOnXXX` - API of Eventuate stays the same (facilitating the new `AbstractActor.Receive` ) but users could also decide to use the new Akka pattern instead of overriding `onXX` and returning the often confusing type of `PartialFunction<Object, BoxedUnit>`.

WDYT?